internal/controller/vrg_volrep.go

```diff
@@ -1467,19 +1474,25 @@ func (v *VRGInstance) validateVRStatus(pvc *corev1.PersistentVolumeClaim, volRep
 	return true
 }
 
-// validateVRValidatedStatus validates that VolumeReplicaion resource was validated.
-// Return 2 booleans
+// validateVRValidatedStatus validates that VolumeReplication resource was validated.
+// Returns 2 values:
 // - validated: true if the condition is true, otherwise false
-// - ok: true if the check was succeesfull, false if the condition is missing, stale, or unknown.
+// - state: condition state
 func (v *VRGInstance) validateVRValidatedStatus(
+	pvc *corev1.PersistentVolumeClaim,
 	volRep *volrep.VolumeReplication,
-) (bool, bool) {
-	conditionMet, errorMsg := isVRConditionMet(volRep, volrep.ConditionValidated, metav1.ConditionTrue)
-	if errorMsg != "" {
-		v.log.Info(fmt.Sprintf("%s (VolRep: %s/%s)", errorMsg, volRep.GetName(), volRep.GetNamespace()))
+) (bool, conditionState) {
+	conditionMet, condState, errorMsg := isVRConditionMet(volRep, volrep.ConditionValidated, metav1.ConditionTrue)
+	if !conditionMet && condState != conditionMissing {
```

Comment:
I would still log the `errorMsg` when the `condState` is `conditionMissing`

Reply:
One issue, since we log the defaultMsg (like we do for other conditions), we will log:

     "VolumeReplication resource not validated"

Which is confusing when using csi-addons < 0.10.0. For example when we Completed=True, which means that everything is good. So I think the best way is to keep the current behavior, log only if the condition exists.